### PR TITLE
IA-4894: upgrade hail to 0.2.130

### PR DESCRIPTION
**Pull request by cpate4**
__Jira ticket__: https://broadworkbench.atlassian.net/browse/IA-4894

### What

- Upgrading hail version to 0.2.130 (latest)

### Why

- Per request

### What to test

- 

### Who tested and where

- [ ] This change is covered by automated tests 
  - _NB: Rerun automation tests on this PR by commenting `jenkins retest` or `jenkins multi-test`._
- [ ] I validated this change 
- [ ] Primary reviewer validated this change 
- [ ] I validated this change in the dev environment

### CODE REVIEW

**Review by LizBaldo — APPROVED** [review]
So in the end AOU will be opening the PR to upgrade their image? I like separating in 2 different PRs anyways :)

You should be good once the integration tests pass

**Review by lucymcnatt — APPROVED** [review]

**Comment by cpate4** [comment]
The [custom image run](https://github.com/DataBiosphere/leonardo/actions/runs/8816723240) has finished ... 
added those changes to this PR as well.

File: http/src/main/resources/reference.conf [hunk 1]
```diff
@@ -110,7 +110,7 @@ dataproc {
 }
 
 gce {
-  customGceImage = "projects/broad-dsp-gcr-public/global/images/leo-gce-image-2024-02-14-15-02-28"
+  customGceImage = "projects/broad-dsp-gcr-public/global/images/leo-gce-image-2024-04-24-12-44-28"
```
Comment (LizBaldo) [inline http/src/main/resources/reference.conf]: Oh, I see, there is no need to do that FYI since the GCE image has not changed, but it does not hurt

**Comment by yonghaoy** [comment]
I was thinking AoU PRs will also be here? https://github.com/DataBiosphere/leonardo/pull/4469
Do we expect another PR for AoU or you want AoU team handles AoU upgrade?

**Comment by cpate4** [comment]
We were [waiting for feedback](https://broadinstitute.slack.com/archives/CNK2A1QP3/p1713465246928409?thread_ts=1712331379.842669&cid=CNK2A1QP3) and were unclear on how to proceed given you had initiated your own PR

**Comment by yonghaoy** [comment]
> We were [waiting for feedback](https://broadinstitute.slack.com/archives/CNK2A1QP3/p1713465246928409?thread_ts=1712331379.842669&cid=CNK2A1QP3) and were unclear on how to proceed given you had initiated your own PR

Ah, sorry did you realize the "feedback" you mentioned is about this Leo config upgrade. I through you were asking the PR in terra-docker which I initialized. 
Usually IA would help us to make Leo version upgrade PR and we coordinate via slack. 
I did not realize things has changed so did not pay too much attension on what exact you were asking in that slack..

**Comment by yonghaoy** [comment]
And about `I like separating in 2 different PRs anyways :)` but that means we will generate cache twice..

**Pull request merged**